### PR TITLE
Add spell check to CI

### PR DESCRIPTION
**Pull request by danieljurek**
Adds spell checking to the `Analyze` build job.

The spell checker has encountered many spelling errors. Some of these errors may be false positives. Teams should go through the errors according to [instructions in the internal wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/406/Spell-Checking).

The issue tracking spelling corrections (which might best be divided into an epic) to track fixing these spelling errors is: https://github.com/Azure/azure-sdk-for-cpp/issues/1277

### CODE REVIEW

**Comment by katmsft** [comment]
Is there a way to add more info for the spell check failures so we could know how to fix them? Now only the file that failed is shown.
Also, is there a way of running the full spell check for all the files before failing on a particular file? This way we can also fix all the failures once CI fails.
Thanks!

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -146,6 +146,12 @@ jobs:
       VcpkgDependencies: curl[winssl] libxml2 nlohmann-json
       VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
     steps:
+      - script: npm i -g cspell
+        displayName: Install cspell
+
+      - script: cspell lint --config cspell.json --no-summary "sdk/${{ parameters.ServiceDirectory }}/**/*"
```
Comment (RickWinter) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:152]: There are lots of spell checkers available.  Is each repro using its own language specific spell checker?  Is there a spell checker we could share across all repos?

**Comment by danieljurek** [comment]
Updated the description at the top. Documentation for how to deal with spelling errors is here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/406/Spell-Checking

**Comment by ahsonkhan** [comment]
> The spell checker has encountered many spelling errors. Some of these errors may be false positives. Teams should go through the errors according to instructions in the internal wiki.

> Updated the description at the top. Documentation for how to deal with spelling errors is here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/406/Spell-Checking

Given this will impact all OSS contributors, please grab relevant parts of that wiki and put it in a publicly accessible location on GitHub (either checked-in as a markdown in this repo, or in a centralized public azure-sdk/engsys repo). I might be mistaken, but I don't see anything in that wiki doc that has internal build system related info or mention of artifacts that can't be public.

**Comment by danieljurek** [comment]
Closing this to keep the main repo clean. We can run testing from my fork.

**Pull request closed without merging**